src/types/guilds/create_guild_channel.ts

```diff
@@ -0,0 +1,32 @@
+import { DiscordChannelTypes } from "../channels/channel_types.ts";
+import { Overwrite } from "../channels/overwrite.ts";
+import { SnakeCaseProps } from "../util.ts";
+
+export interface CreateGuildChannel {
+  /** Channel name (2-100 characters) */
+  name: string;
+  /** The type of channel */
+  type?: DiscordChannelTypes;
+  /** Channel topic (0-1024 characters) */
+  topic?: string;
+  /** The bitrate (in bits) of the voice channel (voice only) */
+  bitrate?: number;
+  /** The user limit of the voice channel (voice only) */
+  userLimit?: number;
+  /** Amount of seconds a user has to wait before sending another message (0-21600); bots, as well as users with the permission manageMessages or manageChannel, are unaffected */
+  rateLimitPerUser?: number;
+  /** Sorting position of the channel */
+  position?: number;
+  /** The channel's permission overwrites */
+  permissionOverwrites?: Overwrite[];
+  /** Id of the parent category for a channel */
+  parentId?: string;
+  /** Whether the channel is nsfw */
+  nsfw?: boolean;
+}
+
+/** https://discord.com/developers/docs/resources/guild#create-guild-channel */
+export interface DiscordCreateGuildChannel
+  extends SnakeCaseProps<Omit<CreateGuildChannel, "permissionOverwrites">> {
+  permission_overwrites: DiscordOverwrite[];
+}
```

Comment:
Why is this done, may I ask?

Reply:
Because we use an permission array and not bits for the overwrites.